src/ddsrt/include/dds/ddsrt/random.h

```diff
@@ -22,23 +31,102 @@ extern "C" {
 
 #define DDSRT_MT19937_N 624
 
+/** @brief A composite seed consisting of multiple smaller seeds */
 typedef struct ddsrt_prng_seed {
   uint32_t key[8];
 } ddsrt_prng_seed_t;
 
+/** @brief The random number generator's state object */
 typedef struct ddsrt_prng {
   uint32_t mt[DDSRT_MT19937_N];
   uint32_t mti;
 } ddsrt_prng_t;
 
-void ddsrt_random_init (void);
-void ddsrt_random_fini (void);
-void ddsrt_prng_init_simple (ddsrt_prng_t *prng, uint32_t seed);
-bool ddsrt_prng_makeseed (struct ddsrt_prng_seed *seed);
-void ddsrt_prng_init (ddsrt_prng_t *prng, const struct ddsrt_prng_seed *seed);
-uint32_t ddsrt_prng_random (ddsrt_prng_t *prng);
-size_t ddsrt_prng_random_name(ddsrt_prng_t *prng, char* output, size_t output_size);
+/**
+ * @brief Initialize the pseudo random number generator
```

Comment:
This is not initialising "the PRNG" but rather initialising "the *global* PRNG", the one that `ddsrt_random` uses. You can safely init/use your own PRNGs.

I think it would be wise to make this a bit more obvious in the `@brief` description.